bql/semantic/expression.go

```diff
@@ -81,24 +81,60 @@ func (o OP) String() string {
 	}
 }
 
+// formatCell formats a given cell into a trimmed comparable string
+func formatCell(c *table.Cell) string {
+	if c.L != nil {
+		return strings.TrimSpace(c.L.ToComparableString())
+	}
+	return strings.TrimSpace(c.String())
+}
+
 // evaluationNode represents the internal representation of one expression.
 type evaluationNode struct {
-	op OP
+	op OP // operation
 
-	lB string
-	rB string
+	lB string // left binding
+	rB string // right binding
 }
 
-// format cell to a trimmed comparable string
-func formatCell(c *table.Cell) string {
-	if c.L != nil {
-		return strings.TrimSpace(c.L.ToComparableString())
+// Evaluate the expression.
+func (e *evaluationNode) Evaluate(r table.Row) (bool, error) {
+	// Binary evaluation
+	eval := func() (*table.Cell, *table.Cell, error) {
+		var (
+			eL, eR *table.Cell
+			ok     bool
+		)
+		eL, ok = r[e.lB]
+		if !ok {
+			return nil, nil, fmt.Errorf("comparison operations require the binding value for %q for row %q to exist", e.lB, r)
+		}
+		eR, ok = r[e.rB]
+		if !ok {
+			return nil, nil, fmt.Errorf("comparison operations require the binding value for %q for row %q to exist", e.rB, r)
+		}
+		return eL, eR, nil
+	}
+
+	eL, eR, err := eval()
+	if err != nil {
+		return false, err
+	}
+	csEL, csER := formatCell(eL), formatCell(eR)
+	switch e.op {
+	case EQ:
+		return reflect.DeepEqual(csEL, csER), nil
+	case LT:
+		return csEL < csER, nil
+	case GT:
+		return csEL > csER, nil
+	default:
+		return false, fmt.Errorf("boolean evaluation require a boolen operation; found %q instead", e.op)
 	}
-	return strings.TrimSpace(c.String())
 }
 
-// binary evaluation - get value for binding in given row
-func getValue(binding string, r table.Row) (*table.Cell, error) {
+// getCellFromRow retrives the value of a binding from a given row
```

Comment:
Golang usually does not recommend prefixing  functions with get because it is argued that it adds nothing to the readability of the function.